File: introduction/ITP1_4_C-Simple Calculator.scala
object Main {
  def main(args: Array[String]):Unit = {
    for(line <- scala.io.Source.stdin.getLines() if line.size > 0) {
      val str = line.split(" ")
      if(!str.contains("?")) {
        println(calc(str(0).toInt, str(1).toString, str(2).toInt))
      }
    }
  }
  def calc(a:Int, op:String, b:Int):Int = {
    op match {
      case "+" => a + b
      case "-" => a - b
      case "*" => a * b
      case "/" => a / b
    }
  }
}

File: introduction/ITP1_4_C-Simple Calculator.js
process.stdin.resume();
process.stdin.setEncoding('utf8');
process.stdin.on('data', function (chunk) {
  var lines = chunk.toString().split('\n');
  for(var i = 0; i < lines.length; i++) {
    var line = lines[i];
    if(line.match(/\?/)) {
      break;
    }
    console.log(Math.floor(eval(line)));
  }
});
